code/modules/network/commands/directory_commands.dm

```diff
@@ -0,0 +1,332 @@
+	feedback = "<span class='notice'>No Subnetwork Available.</span>"
+	return feed(H)
+
+
+/datum/network_command/get_id
+	trigger = "get_id"
+	info = "Usage: \"get_id\". Returns the id of this network."
+
+/datum/network_command/get_id/execute(datum/network/N, datum/network_argument/A, obj/item/device/hacktool/H)
+	..()
+	feedback = "<span class='notice'>Network ID: [N.id]</span>"
+	return feed(H)
+
+
+/datum/network_command/get_key
+	trigger = "get_key"
+	info = "Usage: \"get_key\". Returns the network encryption key."
+
+/datum/network_command/get_key/execute(datum/network/N, datum/network_argument/A, obj/item/device/hacktool/H)
+	..()
+	feedback = "<span_class='notice'>Encryption Key: [N.password]</span>"
+	return feed(H)
+
+
+/datum/network_command/get_hidden
+	trigger = "get_hidden"
+	info = "Usage: \"get_hidden\". Returns whether or not this network is protected from probing."
+
+/datum/network_command/get_hidden/execute(datum/network/N, datum/network_argument/A, obj/item/device/hacktool/H)
+	..()
+	feedback = "<span class='notice'>Network Stealthed: [N.stealth ? "TRUE" : "FALSE"]</span>"
+	return feed(H)
+
+
+/datum/network_command/get_local
+	trigger = "get_local"
+	info = "Usage: \"get_local\". Returns whether or not this network can be accessed from anywhere on the station."
+
+/datum/network_command/get_wireless/execute(datum/network/N, datum/network_argument/A, obj/item/device/hacktool/H)
+	..()
+	feedback = "<span class='notice'>Local Network: [N.wireless ? "FALSE" : "TRUE"]</span>"
+	return feed(H)
+
+
+/datum/network_command/rootnet
+	trigger = "rootnet"
+	info = "Usage: \"rootnet -{encryption_key}\". Attempts to connect to the root network. If no encryption key is provided, current networks encryption key will be used."
+
+/datum/network_command/rootnet/execute(datum/network/N, datum/network_argument/A, obj/item/device/hacktool/H)
+	..()
+	if(!N.root || (N.root.stealth && !ishacktool(H)))
+		feedback = "<span class'notice'>No Root Network.</span>"
+		return feed(H)
+	if(!N.root.password || N.root.password == N.password || N.root.password == A.args[1])
+		connect(N.root, H)
+		return feed(H)
+	if(N.root.password && ishacktool(H) && A.args[1] == "b")
+		feedback = "<span class='warning'>Performing bruteforce hack on [N.root.id].</span>"
+		N.root.execute("security -brute -[N.id]", H) // Trigger the root's security. Loudly.
+		connect(N.root, H)
+		return feed(H)
+	feedback = "<span class='warning'>Encryption Key Mismatch.</span>"
+	N.root.execute("security -alert -[N.id]", H) // Trigger the root's security. Quietly.
+	return feed(H)
+
+
+/datum/network_command/subnet
+	trigger = "subnet"
+	info = "Usage: \"subnet -{target}\". Attempts to connect to the target subnetwork of this network if it exists. If there is only one subnetwork, the target is implicit."
+
+/datum/network_command/subnet/execute(datum/network/N, datum/network_argument/A, obj/item/device/hacktool/H)
+	..()
+	var/datum/network/SN
+	if(A.args[1] && N.sub.len >= 1)
```

Comment:
a lot of it was kinda hastily coded at 1am, i've gone over it like 4 times so far cleaning and fixing stuff up. Thanks for pointing some of the stuff i missed out :)